### PR TITLE
feat(vitest): add Vite plugin for Svelte browser import and autocleanup

### LINKED ISSUE #359: Feature proposal: Vite plugin
## Overview

This library requires three bits of setup to work correctly with Vitest

1. Browser bundle resolution
2. Testing library cleanup
3. `jest-dom` matchers (optional, but pretty much everyone wants it)

Setting these up is a bit fiddly and results in inbound issues in this repository

## Proposal

The most effective method at resolving (1) appears to be setting up a simple Vite plugin. A Vite plugin would also be capable of configuring (2) and maybe even (3). I think we should add a Vite plugin as an export to this library.

Usage (e.g. with SvelteKit)

```ts
// vite.config.ts
import { defineConfig } from 'vite'
import { svelteTesting } from '@testing-library/svelte/vite-plugin'
import { svelteKit } from '@sveltejs/kit/vite'

export default defineConfig({
  plugins: [svelteKit(), svelteTesting()],
})
```

```ts
// @testing-library/svelte/vite-plugin
export const svelteTesting = ({
  resolveBrowser = true,
  autoCleanup = true,
  extendExpect = true,
}) => ...
```

## Related issues

- #222 
- #358

### PR DESCRIPTION
**Pull request by mcous**
This PR adds a Vite plugin export to more reliably configure STL with Vitest:

- Ensure browser Svelte imported instead of node via `resolve.conditions`
- Add auto cleanup without global "afterEach" to `test.setupFiles` automatically

Closes #359

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,10 @@
     },
     "./vitest": {
       "default": "./src/vitest.js"
+    },
+    "./vite-plugin": {
```
Comment (yanick) [inline package.json:18]: I'd just call the file `vite` instead of `vite-plugin`.

**Review by yanick — APPROVED** [review]
Pretty good idea! Should be mentioned in the README, probably?

**Comment by mcous** [comment]
Pushed a couple `package.json` fixes based usage with an actual test suite. Docs PR is up, this one is ready to go!

File: src/vite.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+/**
+ * Add auto-cleanup file to Vitest's setup files.
+ *
+ * @param {import('vitest/config').UserConfig} config
+ */
+const addAutoCleanup = (config) => {
+  const test = config.test ?? {}
+  const setupFiles = test.setupFiles ?? []
```
Comment (mcous) [inline src/vite.js:65]: `setupFiles` may be `string | string[]`, need to account for that case

**Pull request merged**